modules/apps/staging/staging-test/src/testIntegration/java/com/liferay/staging/test/StagingDataPortletPreferencesTest.java

```diff
@@ -309,6 +319,37 @@ protected String publishLayoutWithDisplayPortlet(
 
 	protected PortletPreferences livePortletPreferences;
 
+	private void _setupDDLDisplayPortlet() throws Exception {
```

Comment:
I think this can be simplified like this: https://github.com/liferay/liferay-portal/blob/2a14d490f928a14820c311635d9d2143eb6c637f/modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/test/util/APIApplicationTestUtil.java#L28-L46 with a ServiceTracker and we would not need to use a latch. But I leave it up to you @peerkar to decide.

Reply:
Maybe I am wrong, but here we need to wait for the DDLDisplayPortlet to be registered (or the bundle), but with your solution I think the open method will continue the execution even though there is nothing registered, right?

Reply:
Yeah that was the idea, a bad one. Now it only waits for the portlet, stops right there if not available and gives some information.